GovtOrganization Selector

Jacobjeevan opened the pull request:
## Proposed Changes

- Fixes #9761 
 - Redesigning the selector to render auto complete drop downs that are editable at any time, rather than rendering already selected items with a trash icon.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Introduced a new custom hook for managing government organization data
	- Enhanced organization selection component with improved filtering and selection logic
	- Added dynamic organization level selection with loading indicators

- **Improvements**
	- Refactored organization selection to use a more modular, component-driven approach
	- Improved user experience during data retrieval with loading states

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jacobjeevan committed:
Switching selector to autocomplete and using similar logic to organizationFilter

Jacobjeevan committed:
minor tweaks and edits

Jacobjeevan committed:
added loader

nihal467 committed:
Merge branch 'develop' into issues/9761-govtOrganization

nihal467 committed:
Merge branch 'develop' into issues/9761-govtOrganization

nihal467 commented:
PR looks good to me, need to fix the cypress

khavinshankar commented on src/hooks/useGovtOrganizationLevel.ts:
```diff
@@ -0,0 +1,79 @@
+}: UseGovtOrganizationLevelProps) {
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const headers = authToken
+    ? {
+        headers: {
+          Authorization: `Bearer ${authToken}`,
+        },
+      }
+    : {};
```
we use auth tokens everwhere, so this is fine

khavinshankar commented:
Let's fix the cypress issue in a separate PR.

cc: @Jacobjeevan @nihal467

nihal467 commented:
![image](https://github.com/user-attachments/assets/1681abfb-4b4d-4d46-9c07-e77c049247ae)

when you type to try aluva in the local body dropdown, its goes into loading state in between, seem like a state management issue, can you look into it

Jacobjeevan commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/405664907-1681abfb-4b4d-4d46-9c07-e77c049247ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc1NjY2NDMsIm5iZiI6MTczNzU2NjM0MywicGF0aCI6Ii81NzA1NTk5OC80MDU2NjQ5MDctMTY4MWFiZmItNGI0ZC00ZDQ2LTljMDctZTc3YzA0OTI0N2FlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIyVDE3MTkwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmMjUxYzNlNzliYTU0ODM0M2ZkMjFkMzVkYzIzMmI3OTg3YjA4YmIwNTk4NDZkZjMzODY5NWMyOTNlN2YwZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B1ieo4J91p5jiU_Y66_Qh8PrRHGZr8snDg__LWw2wN4)
> 
> when you type to try aluva in the local body dropdown, its goes into loading state in between, seem like a state management issue, can you look into it

I had added a spinner, funnily missed testing the search portion after adding the spinner. Anyways, adjusted to display spinner alongside the dropdown, rather than instead of.

#10105
